### DISCUSSION TITLE
Get the integer part of the variables or get the current values of the variables

### DISCUSSION THREAD

**Discussion by tedzit** [coin-or/CppAD#125] [Q&A]
Hello everyone!
I am new to CppAD. I find it very interesting and I would like to use it to code the command of mobile robot.
### Context:
The vector x represents the variables of the speed and the breaking angle of the N next horizons time.
The objective function based on the evolution model of the robot works good.
But the constrained function which look the path where the robot will pass in a certain horizon time do not work.
For this function I need to get the values of the cells in an occupancy grid where the robot will pass over.
The constrained function is basically a sum of these values and has to be under a threshold value.
### Problem:
In order to get the coordinates of the cells I need to get tan integer type from positions of the robot given by the evolution of the robot (and so the vector of variables x).
However, I can not do this. I do not find a function ceil or floor or round to work with AD<double> type.
A way to go over it, would be to get the current values of the vector x, but it always give me zero when I use the function Value(Var2Par(x[i])).

If you have any tips to give to me it will be very kind of you.
Best Regards,
Ted

**Comment by bradbell** [comment]
You could the following feature
https://coin-or.github.io/CppAD/doc/discrete.htm
to define a floor or ceil function.

One thing to remember is that the derivative will evaluate at zero at point where the function is not continuous and the derivative is actually a Dirac Delta distribution.
